src/sentry/auth/elevated_mode.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+
+
+class ElevatedMode(ABC):
+    @staticmethod
+    @abstractmethod
+    def _needs_validation():
```

Comment:
I'm not sure if this function is needed. In superuser, it skips validating the modal if:
1. self-hosted
2. some conf setting is True
3. otherwise, you pull from some other conf setting

Reply:
for staff, in the spec i wrote that we skip validation if it's self hosted

Reply:
Can leave _needs_validation in for now then